From bbf6dfa8a691d20ee85de2e7d61aed7053dbd4de Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Tue, 30 Mar 2021 16:45:58 +0300
Subject: [PATCH] [SW-36011] boot_info: add flush api in boot info

Add capability to flush boot info into SP SRAM
during preboot runtime (not only when loading
u-boot), this is required in case the device gets
reset before loading u-boot, which can result in
loss of the SP SRAM info.

Change-Id: I5ebb9cff20723cc99ee7497fa8dea85db2a679ed
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/preboot/boot_info.c b/preboot/boot_info.c
index 0bfcd8b..17a1b8a 100644
--- a/preboot/boot_info.c
+++ b/preboot/boot_info.c
@@ -60,6 +60,7 @@
 	const u32 *src;
 	u32 *dst;
 	int i, j;
+	struct hl_boot_info_desc descs[MAX_INFO_DESCS];
 
 	if (!boot_info.ptrs.count)
 		return;
@@ -73,9 +74,10 @@
 
 	dst = offset2ptr(offset, BOOT_INFO_BASE);
 
+	memcpy(descs, boot_info.descs, sizeof(descs));
 	/* Store data */
 	for (i = 0; i < boot_info.ptrs.count; i++) {
-		desc = &boot_info.descs[i];
+		desc = &descs[i];
 		src = offset2ptr(desc->offset, PREBOOT_TEXT_BASE);
 		sha256_csum((u8 *)src, desc->size, desc->chksum);
 		copy_sz = ROUND_UP4(desc->size);
@@ -92,7 +94,7 @@
 		(sizeof(u32)) * boot_info.ptrs.count;
 	dst = offset2ptr(offset, BOOT_INFO_BASE);
 	for (i = 0; i < boot_info.ptrs.count; i++) {
-		src = (u32 *)&boot_info.descs[i];
+		src = (u32 *)&descs[i];
 		copy_sz = ROUND_UP4(sizeof(struct hl_boot_info_desc));
 
 		for (j = 0; j < copy_sz / sizeof(u32); j++)
